deploy airship ci: create combined inventory file

aagate opened the pull request:
deploy airship ci: create combined inventory file

- added an example hosts_combined_example.yml as a proposal to combine "deploy_osh"
  "deploy_airship" inventory files and create a
  combined inventory file that can work for both osh and airship deployment.

- updated examples/inventory/hosts.yml with hosts that are necessary
  or airship deployment.

- extend inventory hosts.yml tasks now also sets following
  host groups required for airship deployment

  'airship-openstack-control-workers' to two caasp worker nodes
  'airship-ucp-worker' to two caasp worker nodes
  'airship-openstack-compute-workers' to third worker node
  'airship-kube-system-workers' to two caasp worker nodes

- switched from 'osh-deployer' host group to 'soc-deployer' in all playbooks.

- switched from  'airship-deployer' host group to 'soc-deployer' in all playbooks.
  * since a deployer can be used for both osh and airship deployment

- switched from 'airship-workers' to 'caasp-workers' in all playbooks.
  * since they are the identical there is no need to have two different
    host groups

aagate commented on examples/workdir/inventory/hosts_combined_example.yml:
```diff
@@ -0,0 +1,78 @@
+---
```
Note: hosts_combined_example file is created for review purposes only and
  includes following proposed changes:
   * added "airship-openstack-control-workers", "airship-ucp-workers"
     and "airship-openstack-compute-workers"  host groups that are
     necessary for airship deployment.
     - for CI,
        "airship-openstack-control-workers": set hosts to first two workers
        "airship-ucp-workers": set hosts to first two workers
        "airship-openstack-compute-workers": set hosts to third worker
   * rename 'osh-deployer' to 'socok8s-deployer'
     - since deployer can be used for osh and airship deployment
   * added "all" and "airship-workers" host groups
   * Questions:
     Q1: does deploy_osh role make use of something similar to
         "airship-openstack-control-workers"  or
         "airship-openstack-compute-workers" ? Do they need to be renamed?

evrardjp commented on examples/workdir/inventory/hosts.yml:
```diff
@@ -18,3 +18,23 @@ osh-deployer:
+
+airship-ucp-workers:
+
+all:
```
all is a meta group and should not be redefined.

aagate replied:
Thx! Will remove

evrardjp commented on examples/workdir/inventory/hosts.yml:
```diff
@@ -18,3 +18,23 @@ osh-deployer:
+
+airship-openstack-control-workers:
+
+airship-ucp-workers:
```
Why not reusing caasp workers?

aagate replied:
This can be a subset of the total available caasp workers.

evrardjp replied:
LGTM then

evrardjp commented on examples/workdir/inventory/hosts_combined_example.yml:
```diff
@@ -0,0 +1,78 @@
+      ansible_host: 10.86.3.220
+      primary: yes
+
+all:
```
To be removed.

aagate replied:
Thx! Will remove

evrardjp commented on playbooks/openstack-caasp_instance.yml:
```diff
@@ -174,6 +174,16 @@
                 hosts: "{{ masterinventory }}"
               caasp-workers:
                 hosts: "{{ workerinventory }}"
+              # TODO(aagate): set hosts to first two workers, move to airship_configure_caasp role
```
if these are the same, why redefining groups?

evrardjp commented on run.sh:
```diff
@@ -72,6 +72,12 @@ case "$deployment_action" in
         build_images
         deploy_osh
         ;;
+    "setup_airship")
```
Great to add this, can this be in a different commit though?

aagate replied:
> Great to add this, can this be in a different commit though?

To test setup_airship, I have to update the inventory hosts file and add groups that airship requires, so the two changes kind of go hand in hand. I could add this in a separate commit but wont be able to test it.

aagate replied:
Went ahead and created separate PR to add 'setup_airship' as per your suggestion. This PR is again getting too long and might be good to just restrict it to inventory changes.

The other PR to add setup_airship is https://github.com/SUSE-Cloud/socok8s/pull/165 I will rebase this once that PR is merged.

aagate replied:
Done. #165 was merged. Rebased this PR.

aagate commented:
> Could you clarify the group definition's purposes?

Added explanation on the host groups

> Could you clarify the group definition's purposes?

Added more info describing different control plane groups and their use

evrardjp commented on playbooks/generic-clean_airship.yml:
```diff
@@ -15,13 +15,13 @@
       script: "{{ playbook_dir }}/../script_library/clean-airship.sh {{ clean_action | default('') }}"
 
-- hosts: airship-workers
+- hosts: caasp-workers
```
I think we should make this file not only a airship thing, but a "clean my caasp setup of all those pesky helm charts" ;) 

For a different commit though.

aagate commented:
rebased

evrardjp commented:
Sensu stricto, the commit message should be updated, not to list what was done, but explain why it was done.

For example, the move from osh-deployer to soc-deployer can be explained with the fact this node can serve on both, and there is no reason to duplicate this in the inventory.

aagate commented:
> Sensu stricto, the commit message should be updated, not to list what was done, but explain why it was done.
> 
> For example, the move from osh-deployer to soc-deployer can be explained with the fact this node can serve on both, and there is no reason to duplicate this in the inventory.

Its been explained in the note here https://github.com/SUSE-Cloud/socok8s/pull/148#discussion_r266698688 I can add some of those points to the commit message too

aagate commented:
Updated commit and PR message, added reason for making changes to each point

sgrasley commented on examples/workdir/inventory/hosts_combined_example.yml:
```diff
@@ -0,0 +1,76 @@
+      ansible_host: 10.86.3.69
+  vars:
+    ansible_user: root
+ses_nodes:
```
ses_nodes has caused errors in airship deployments. Has this been resolved ?

aagate replied:
What kind of errors, are they format related i.e  specifying one node vs specifying multiple nodes?. I do have ses_nodes (single node) in my manual deployment, which seemed to work fine. From airship CI point of view, it will use a single node ses deployment too.  I haven't yet succeeded in getting all the way through a complete aiship deployment, still in deploy fix cycle.

sgrasley replied:
Last time I ran run.sh deploy_airship with ses_nodes in my inventory it tried to update docker on ses_node and failed.  If it is still a problem, it is not something for this review. Just wanted to bring it up.

evrardjp commented:
I am testing this today.

evrardjp commented:
nit: I have seen a discrepency: The inventory file is still named osh where it could be deployer. But this can be improved later.

Still testing it, to see if the deploy_osh step works.
